app/src/main/java/com/github/dedis/student20_pop/ui/HomeFragment.java

```diff
@@ -18,11 +31,106 @@
+    private ArrayList<Lao> laos;
+    //Now: for testing, later: retrieve from Person
+    private Keys keys = new Keys();
+    private String myPublicKey = keys.getPublicKey();
+
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         // TODO: retrieve list of LAOs from backend and display them
 
-        return inflater.inflate(R.layout.fragment_home, container, false);
+        View view = inflater.inflate(R.layout.fragment_home, container, false);
+
+        laos = getLaos();
+
+        LinearLayout welcome = view.findViewById(R.id.welcome_screen);
+        LinearLayout list = view.findViewById(R.id.list_screen);
+
+        if (laos.isEmpty()) {
+            welcome.setVisibility(View.VISIBLE);
+            list.setVisibility(View.GONE);
+        }else{
+            welcome.setVisibility(View.GONE);
+            list.setVisibility(View.VISIBLE);
+        }
+        ListView laosListView = view.findViewById(R.id.lao_list);
+        laosListView.setAdapter(new LaoListAdapter(this.getContext()));
+        return view;
+    }
+
+    private class LaoListAdapter extends BaseAdapter {
+        private Context context;
+
+        public LaoListAdapter(Context context){
+            this.context = context;
+        }
+
+        @Override
+        public int getCount() {
+            return laos.size();
+        }
+
+        @Override
+        public Object getItem(int position) {
+            return laos.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            if (convertView == null) {
+                LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+                convertView = inflater.inflate(R.layout.layout_lao_home, null);
+            }
+            Lao lao = laos.get(position);
+            ((TextView) convertView.findViewById(R.id.lao_name)).setText(lao.getName());
+            ((TextView) convertView.findViewById(R.id.date)).setText("last interacted with ...");
+            boolean isOrganizer = lao.getOrganizer().equals(myPublicKey);
+            if (isOrganizer){
+                ((TextView) convertView.findViewById(R.id.role)).setText(R.string.organizer);
+            }else {
+                ((TextView) convertView.findViewById(R.id.role)).setText(R.string.attendee);
+            }
+            convertView.setOnClickListener(clicked -> {
+               if (isOrganizer){
+                   startOrganizerUI(lao);
+               }else {
+                   startAttendeeUI(lao);
+               }
+            });
+            return convertView;
+        }
+    }
+
+    private ArrayList<Lao> getLaos(){
+        //Now: for testing
```

Comment:
What does this test?

Reply:
It tests the UI with an actual list of LAO's until we implement a way to get real LAO's from the backend